iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/plugin/SpringConfigSensor.java

```diff
@@ -0,0 +1,125 @@
+  private static final Set<String> EXCLUDED_PROFILES = Set.of("dev", "test");
+
+  public SpringConfigSensor(SonarRuntime sonarRuntime, FileLinesContextFactory fileLinesContextFactory, NoSonarFilter noSonarFilter) {
+    // The Java language is registered by the sonar-java plugin. However, for the sensor we only need language key and name, and don't need to
+    // rely on the SQ extension.
+    // Mechanisms of dependency injection between SQ and SL can differ, and the `org.sonar.plugins.java.Java` language is available only in the
+    // `sonar-java` plugin, so it's not possible to inject it here.
+    // That's why the sensor is hardcoding key and name and not providing a `Language` object.
+    super(sonarRuntime, fileLinesContextFactory, noSonarFilter, null);
+  }
+
+  @Override
+  protected String languageName() {
+    return "Java";
+  }
+
+  @Override
+  public void describe(SensorDescriptor descriptor) {
+    // Do not define the sensor only on Java language, because Spring configuration files are not assigned to it.
+    descriptor.name(SENSOR_NAME);
+
+    // The sensor shouldn't call `processFilesIndependently()`, because if a default Spring profile is defined in another file,
+    // it should still be loaded.
+  }
+
+  @Override
+  protected TreeParser<Tree> treeParser() {
+    return new SpringConfigParser();
+  }
+
+  @Override
+  protected FilePredicate mainFilePredicate(SensorContext sensorContext) {
+    var fileSystem = sensorContext.fileSystem();
+    var config = sensorContext.config();
+
+    var patterns = getFilePatterns(config);
+    return fileSystem.predicates().and(
+      fileSystem.predicates().matchesPathPatterns(patterns),
+      new ProfileNameFilePredicate());
+  }
+
+  @Override
+  protected String repositoryKey() {
+    return SpringConfigExtension.REPOSITORY_KEY;
+  }
+
+  @Override
+  protected List<TreeVisitor<InputFileContext>> visitors(SensorContext sensorContext, DurationStatistics statistics) {
+    // TODO: SONARIAC-1437 Implement metrics and highlighting visitors for .properties files
+    return List.of();
+  }
+
+  @Override
+  protected String getActivationSettingKey() {
+    return SpringConfigSettings.ACTIVATION_KEY;
+  }
+
+  static String[] getFilePatterns(Configuration config) {
+    var patterns = Arrays.stream(config.getStringArray(SpringConfigSettings.FILE_PATTERNS_KEY))
+      .filter(s -> !s.isBlank())
+      .toArray(String[]::new);
+    if (patterns.length == 0) {
+      patterns = SpringConfigSettings.FILE_PATTERNS_DEFAULT_VALUE.split(",");
+    }
+    return patterns;
+  }
+
+  private static class ProfileNameFilePredicate implements FilePredicate {
+    private static final Pattern SPRING_CONFIG_FILENAME_PATTERN = Pattern.compile("application(?:-(?<name>[^.]++))?+.(properties|yaml|yml)");
+
+    @Override
+    public boolean apply(InputFile inputFile) {
+      var matcher = SPRING_CONFIG_FILENAME_PATTERN.matcher(inputFile.filename());
+      if (matcher.find()) {
```

Comment:
Why use `find()` instead of matches? Are we expecting a pre or suffix on the filename?